src/coreclr/vm/stringliteralmap.cpp

```diff
@@ -489,7 +489,7 @@ StringLiteralEntry *GlobalStringLiteralMap::AddStringLiteral(EEStringData *pStri
     {
     PinnedHeapHandleBlockHolder pStrObj(&m_PinnedHeapHandleTable,1);
     // Create the COM+ string object.
-    STRINGREF strObj = AllocateStringObject(pStringData);
+    STRINGREF strObj = AllocateStringObject(pStringData, true); // TODO: don't use POH for collectible assemblies?
```

Comment:
I do not a have good sense for how much we should try to avoid fragmentation of POH. It may be ok to allocate these on POH even for the collectible assemblies, at least to start with. @Maoni0 @cshung Any thoughts?

Reply:
Is it possible that by frequently loading and unloading collectible assemblies, these string literals turn into lots of small holes in the POH that we effectively cannot reuse? If so, I think it is better for us to not allocating them on the POH.